main.go

```diff
@@ -120,10 +136,15 @@ func main() {
 		os.Exit(1)
 	}
 
-	metricsServer := osdmetrics.NewBuilder(config.OperatorNamespace, "localmetrics-"+config.OperatorName).
+	ns, err := k8sutil.GetOperatorNamespaceEnv()
+	if err != nil {
+		os.Exit(1)
```

Comment:
You may have already figured it out, but the issue with the current PR is that `OPERATOR_NAMESPACE` is not an environment variable currently known to the operator, so this exits.

I'd take this as an opportunity to unify this bit of code and the earlier bit of code which also checks for the operator namespace prior to creating the lock. I also think it's fine to lean back on a known default (`openshift-rbac-permissions`) if we can't detect the namespace. Several other operators are doing similar.